src/hardware/mixer.cpp

```diff
@@ -1541,6 +1542,23 @@ static void MIXER_MixData(const int frames_requested)
 		pos = (pos + 1) & MIXER_BUFMASK;
 	}
 
+	// Apply compressor to the master output as the very last step
+	pos = start_pos;
+
+	for (work_index_t i = 0; i < frames_added; ++i) {
+		AudioFrame frame = {mixer.work[pos][0], mixer.work[pos][1]};
+
+		frame.left /= INT16_MAX;
+		frame.right /= INT16_MAX;
+
+		frame = mixer.compressor.Process(frame);
+
+		mixer.work[pos][0] = frame.left * INT16_MAX;
+		mixer.work[pos][1] = frame.right * INT16_MAX;
```

Comment:
The Compressor operates on "percentage" samples, which makes these pre-and-post `INT16_MAX` scale down and up feel like the Compressor isn't designed to (fully) suit our needs.  

Maybe the Compressor could be informed of the stream intended sample maximums, and we could pass `INT16_MAX` into `Config(...)`, and have the compressor scale for us in and out?

Reply:
Sure, moving the scaling into the compressor code is a good idea. The scaling still needs to happen as the `exp` functions need the input to be in the (-1.0, 1.0) range.